fivetran_provider_async/operators.py

```diff
@@ -87,11 +102,20 @@ def execute(self, context: Context) -> None | str:
         hook.prep_connector(self.connector_id, self.schedule_type)
         last_sync = hook.start_fivetran_sync(self.connector_id)
 
-        if not self.deferrable:
+        if not self.wait_for_completion:
             return last_sync
+
+        if not self.deferrable:
+            self._wait_synchronously(pendulum.parse(last_sync))  # type: ignore[arg-type]
+            return None
```

Comment:
I think `return` might make more 
```suggestion
            return
```